src/go/rdctl/cmd/listSettings.go

```diff
@@ -40,10 +63,63 @@ var listSettingsCmd = &cobra.Command{
 	},
 }
 
+// the reg file format is directly usable only on Windows,
+// but it can be created on any platform for purposes of testing or development
+
 func init() {
 	rootCmd.AddCommand(listSettingsCmd)
+	listSettingsCmd.Flags().StringVarP(&outputSettingsFlags.Format, "output", "", jsonFormat, fmt.Sprintf("output format: %s|%s, default %s", jsonFormat, regFormat, jsonFormat))
+	listSettingsCmd.Flags().StringVarP(&outputSettingsFlags.RegistryHive, "reg-hive", "", "", fmt.Sprintf("registry hive: %s|%s, default %s", reg.HklmRegistryHive, reg.HkcuRegistryHive, reg.HklmRegistryHive))
+	listSettingsCmd.Flags().StringVarP(&outputSettingsFlags.RegistryProfileType, "section", "", "", fmt.Sprintf("registry section: %s|%s, default %s", defaultsRegistrySection, lockedRegistrySection, defaultsRegistrySection))
+}
+
+func validateOutputFormatFlags() error {
+	if outputSettingsFlags.Format != jsonFormat && outputSettingsFlags.Format != regFormat {
+		return fmt.Errorf(`invalid output format of "%s"`, outputSettingsFlags.Format)
+	}
+	if outputSettingsFlags.Format == jsonFormat {
+		if outputSettingsFlags.RegistryHive != "" || outputSettingsFlags.RegistryProfileType != "" {
+			return fmt.Errorf("registry hive and profile can't be specified with json")
+		}
+		return nil
+	}
+	switch strings.ToLower(outputSettingsFlags.RegistryHive) {
+	case reg.HklmRegistryHive, reg.HkcuRegistryHive:
+		outputSettingsFlags.RegistryHive = strings.ToLower(outputSettingsFlags.RegistryHive)
+	case "":
+		outputSettingsFlags.RegistryHive = reg.HklmRegistryHive
+	default:
+		return fmt.Errorf("invalid registry hive of '%s' specified", outputSettingsFlags.RegistryHive)
+	}
+	switch strings.ToLower(outputSettingsFlags.RegistryProfileType) {
+	case defaultsRegistrySection, lockedRegistrySection:
+		outputSettingsFlags.RegistryProfileType = strings.ToLower(outputSettingsFlags.RegistryProfileType)
+	case "":
+		outputSettingsFlags.RegistryProfileType = defaultsRegistrySection
+	default:
+		return fmt.Errorf("invalid registry section of '%s' specified", outputSettingsFlags.RegistryProfileType)
+	}
+	return nil
 }
 
 func getListSettings() ([]byte, error) {
-	return processRequestForUtility(doRequest("GET", versionCommand("", "settings")))
+	err := validateOutputFormatFlags()
+	if err != nil {
+		return nil, err
+	}
+	output, err := processRequestForUtility(doRequest("GET", versionCommand("", "settings")))
+	if err != nil {
+		return nil, err
+	} else if outputSettingsFlags.Format == jsonFormat {
+		return output, nil
+	} else if outputSettingsFlags.Format == regFormat {
+		lines, err := reg.JsonToReg(outputSettingsFlags.RegistryHive, outputSettingsFlags.RegistryProfileType, string(output))
+		if err != nil {
+			return nil, err
+		}
+		return []byte(strings.Join(lines, "\n")), nil
```

Comment:
The caller just cases this back to a string again; maybe make this function return a `string` instead of `[]byte` would make more sense?

Reply:
sure